lib/open_trip_planner_client/schema/leg.ex

```diff
@@ -107,14 +107,38 @@ defmodule OpenTripPlannerClient.Schema.Leg do
   To be grouped together, legs must share these characteristics:
   - Same origin and destination
   - Same :transit_leg value (e.g. walking legs don't get grouped with transit legs)
-  - Same transit route.type, except for rail replacement buses
+  - Same transit mode, or using rail replacement buses
   """
   @spec group_identifier(__MODULE__.t()) :: tuple()
   def group_identifier(%__MODULE__{transit_leg: false} = leg) do
     {:WALK, leg.from.name, leg.to.name}
   end
 
-  def group_identifier(%__MODULE__{route: %Route{desc: desc, type: type}} = leg) do
-    {type, desc == "Rail Replacement Bus", leg.from.name, leg.to.name}
+  def group_identifier(%__MODULE__{agency: %Agency{name: agency_name}} = leg)
+      when agency_name != "MBTA" do
+    {agency_name, leg.from.name, leg.to.name}
+  end
+
+  def group_identifier(%__MODULE__{route: %Route{desc: "Rail Replacement Bus"}} = leg) do
+    {:shuttle, leg.from.name, leg.to.name}
+  end
+
+  def group_identifier(%__MODULE__{route: %Route{type: type}} = leg) when type in [0, 1] do
+    {:subway, leg.from.name, leg.to.name}
```

Comment:
Huh - I'm actually a little surprised that [this trip](https://www.mbta.com/trip-planner?plan=hcQVX3VudXNlZF9kYXRldGltZV90eXBlxADECGRhdGV0aW1lxCAyMDI1LTA0LTIzVDEzOjQwOjQ3Ljg1NDM0MC0wNDowMMQEZnJvbYTECGxhdGl0dWRly0BFLnd43WFwxAlsb25naXR1ZGXLwFHDuwX668TEBG5hbWXECUhheW1hcmtldMQHc3RvcF9pZMQLcGxhY2UtaGFlY2zEBW1vZGVzicQDQlVTxAR0cnVlxAVGRVJSWcQEdHJ1ZcQEUkFJTMQEdHJ1ZcQGU1VCV0FZxAR0cnVlxA5fcGVyc2lzdGVudF9pZMQBMMQLX3VudXNlZF9CVVPEAMQNX3VudXNlZF9GRVJSWcQAxAxfdW51c2VkX1JBSUzEAMQOX3VudXNlZF9TVUJXQVnEAMQCdG-ExAhsYXRpdHVkZcQAxAlsb25naXR1ZGXEAMQEbmFtZcQNTm9ydGggU3RhdGlvbsQHc3RvcF9pZMQA) works, since the Green Line and Orange Line have different `type`'s.

![Screenshot 2025-04-23 at 1 39 43 PM](https://github.com/user-attachments/assets/af3f9274-1937-46ce-a99a-dbfefdae2711)

Reply:
Same! Like, what lol

Reply:
Figured it out! Dotcom [is using](https://github.com/mbta/dotcom/blob/0648cefe02c19cf9ae32caad5265ad0f6d3b0259/lib/dotcom/trip_plan/itinerary_groups.ex#L49-L52) `Routes.Route.type_atom` instead of just plain-old-type, and [that resolves](https://github.com/mbta/dotcom/blob/0648cefe02c19cf9ae32caad5265ad0f6d3b0259/lib/routes/route.ex#L108-L109) both `type: 1` and `type: 0` to `:subway`.

So the reason it worked was because this itinerary grouping code was (slightly) broken, but wasn't being used!